### PR TITLE
[4.0] HTML class naming standard [frontend][com-newsfeeds]

### PR DESCRIPTION
**Pull request by ciar4n**
Pull Request for Issue #15279 .

### Summary of Changes
Adds HTML classes using the following standard to com-tags frontend views.

`ExtensionType-Extension(-SubExtension)(-View)` (Eg. `mod-newsfeeds-category`).

Note: If the default view then `-View` is omitted. If only one SubExtension exists or if the SubExtension matches the Extension then `-SubExtension ` is omitted.

#### Child views and child elements within the views

BEM class naming is adopted.. 

- **Block** (https://en.bem.info/methodology/quick-start/#block): `ExtensionType-Extension(-View)`
- **Element** (https://en.bem.info/methodology/quick-start/#element): Child view/element

`ExtensionType-Extension__Element`.

For B/C, old classes remain. 

### Testing Instructions
Code review.

### Expected result
All works fine

### Actual result
All works fine

### Documentation Changes Required
Yes

### CODE REVIEW

File: components/com_newsfeeds/tmpl/categories/default_items.php [hunk 1]
```diff
@@ -12,14 +12,14 @@
 use Joomla\Component\Newsfeeds\Site\Helper\Route as NewsfeedsHelperRoute;
 
 ?>
-<?php $class = ' class="first"'; ?>
+<?php $class = ' first'; ?>
```
Comment (Quy) [inline components/com_newsfeeds/tmpl/categories/default_items.php]: Fix it like here: https://github.com/joomla/joomla-cms/commit/6f5c635d726f894577075d7622e2205943918924#diff-f30110b1d143f638e2fecb2cb99024af

File: components/com_newsfeeds/tmpl/categories/default_items.php [hunk 2]
```diff
@@ -12,14 +12,14 @@
...
 <?php if ($this->maxLevelcat != 0 && count($this->items[$this->parent->id]) > 0) : ?>
 	<?php foreach ($this->items[$this->parent->id] as $id => $item) : ?>
 		<?php if ($this->params->get('show_empty_categories_cat') || $item->numitems || count($item->getChildren())) : ?>
 			<?php if (!isset($this->items[$this->parent->id][$id + 1])) : ?>
 				<?php $class = ' class="last"'; ?>
```
Comment (Quy) [inline components/com_newsfeeds/tmpl/categories/default_items.php]: Fix

File: components/com_newsfeeds/tmpl/categories/default_items.php (hunk 1 above)
Comment (ciar4n) [reply components/com_newsfeeds/tmpl/categories/default_items.php]: Thank you. Amended.

File: components/com_newsfeeds/tmpl/categories/default_items.php (hunk 2 above)
Comment (Quy) [reply components/com_newsfeeds/tmpl/categories/default_items.php]: Change to `<?php $class = 'last'; ?>`

File: components/com_newsfeeds/tmpl/categories/default_items.php [hunk 3]
```diff
@@ -12,14 +12,14 @@
 use Joomla\Component\Newsfeeds\Site\Helper\Route as NewsfeedsHelperRoute;
 
 ?>
-<?php $class = ' class="first"'; ?>
+<?php $class = 'first'; ?>
 <?php if ($this->maxLevelcat != 0 && count($this->items[$this->parent->id]) > 0) : ?>
 	<?php foreach ($this->items[$this->parent->id] as $id => $item) : ?>
 		<?php if ($this->params->get('show_empty_categories_cat') || $item->numitems || count($item->getChildren())) : ?>
 			<?php if (!isset($this->items[$this->parent->id][$id + 1])) : ?>
 				<?php $class = ' class="last"'; ?>
```
Comment (Quy) [inline components/com_newsfeeds/tmpl/categories/default_items.php]: Change to `<?php $class = 'last'; ?>`

File: components/com_newsfeeds/tmpl/category/default_items.php [hunk 4]
```diff
@@ -14,82 +14,86 @@
...
-			<?php if (($this->params->def('show_pagination', 2) == 1 || ($this->params->get('show_pagination') == 2)) && ($this->pagination->pagesTotal > 1)) : ?>
-				<div class="w-100">
-					<?php if ($this->params->def('show_pagination_results', 1)) : ?>
-						<p class="counter float-right pt-3 pr-2">
-							<?php echo $this->pagination->getPagesCounter(); ?>
-						</p>
+					<?php if ($this->params->get('show_link')) : ?>
+						<?php $link = JStringPunycode::urlToUTF8($item->link); ?>
+						<span class="list float-left">
+								<a href="<?php echo $item->link; ?>">
```
Comment (Quy) [inline components/com_newsfeeds/tmpl/category/default_items.php]: Remove 1 indentation.

File: components/com_newsfeeds/tmpl/category/default_items.php [hunk 5]
```diff
@@ -14,82 +14,86 @@
...
+	                                <?php echo $link; ?>
```
Comment (Quy) [inline components/com_newsfeeds/tmpl/category/default_items.php]: Use tabs. Repeat next line.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on fb42d5e9aa40abb08f77518bbce2a544d9f7523b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20820">issues.joomla.org/tracker/joomla-cms/20820</a>.</sub>

**Pull request merged**